tf_encrypted/keras/engine/base_layer.py

```diff
@@ -135,6 +136,8 @@ def set_weights(self, weights, sess=None):
         sess.run(tfe.assign(w, tfe_weights_pl), feed_dict=fd)
     elif isinstance(weights[0], PondPrivateTensor):
       for i, w in enumerate(self.weights):
+        if not sess.run(KE.is_variable_initialized(weights[i])):
+          sess.run(weights[i].initializer)
```

Comment:
you're only testing that `weights[i]` are private tensors so not guaranteed to have an initializer (only variables do). i'm not sure any initializers should be called here; if the user passes in a private variable then it's his responsibility to ensure that it's already initialized.

what does TF Keras do if a user passes in a variable?

Reply:
users are not allowed to do so in tf.keras -- they restrict to numpy arrays, but that's not enough for us since we want to be able to do so without reconstructing the weights on the device of whoever runs this code

Reply:
agreed about initializers though, let's assume the users have run these on their own whenever needed

Reply:
Sounds good, let's assume, they have been run.